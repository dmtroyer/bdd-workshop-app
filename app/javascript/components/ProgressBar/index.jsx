import React from 'react';
import { Progress } from 'reactstrap';
import PropTypes from 'prop-types';

const ProgressBar = ({ percentComplete }) => {
  return (
    <>
      <div className="text-center" data-testid="profile-progress">
        {`${percentComplete}%`}
      </div>
      <Progress value={percentComplete} />
    </>
  );
};
ProgressBar.propTypes = {
  percentComplete: PropTypes.number.isRequired,
};

export default ProgressBar;
