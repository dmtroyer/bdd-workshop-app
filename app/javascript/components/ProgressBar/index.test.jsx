import React from 'react';
import { shallow } from 'enzyme';
import ProgressBar from '.';

describe('ProgressBar', () => {
  it('Displays the percentComplete', () => {
    const wrapper = shallow(<ProgressBar percentComplete={33} />);
    expect(wrapper).toMatchSnapshot();
  });
});
