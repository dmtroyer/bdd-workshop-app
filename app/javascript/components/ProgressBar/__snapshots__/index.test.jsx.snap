// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar Displays the percentComplete 1`] = `
<Fragment>
  <div
    className="text-center"
    data-testid="profile-progress"
  >
    33%
  </div>
  <Progress
    max={100}
    tag="div"
    value={33}
  />
</Fragment>
`;
